### ISSUE TITLE
Repair error: invalid partition-end, partition  has an active range tombstone

### ISSUE THREAD

**Issue by r4fek** [scylladb/scylladb#19246]
This is Scylla's bug tracker, to be used for reporting bugs only.
If you have a question about Scylla, and not a bug, please ask it in
our mailing-list at scylladb-dev@googlegroups.com or in our slack channel.

- [x] I have read the disclaimer above, and I am reporting a suspected malfunction in Scylla.

*Installation details*
Scylla version (or git commit hash): 5.4.6-0.20240418.10f137e367e3-1
Cluster size: 3x8 + 1x4 nodes (4 dcs, 28 nodes in total)
OS (RHEL/CentOS/Ubuntu/AWS AMI): Debian 11.9

I have the following error during the repair on every node in one of my dcs:

```
$ nodetool repair accounts refresh_token -pr -local

cze 12 05:53:03 ams-n28-06-12 scylla[4036370]:  [shard  6:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.3.64, keyspace=accounts, table=refresh_token, range=(-9110052604093903749,-9105776348719089090], error=seastar::rpc::remote_verb_error ([validator 0x63001510f820 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1713875097439963, deletion_time=1713875097})
cze 12 05:53:03 ams-n28-06-12 scylla[4036370]:  [shard  6:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=6, keyspace=accounts, cf=refresh_token, range=(-9110052604093903749,-9105776348719089090], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x63001510f820 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1713875097439963, deletion_time=1713875097})
cze 12 05:53:04 ams-n28-06-12 scylla[4036370]:  [shard  7:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d006d92320 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:04 ams-n28-06-12 scylla[4036370]:  [shard  5:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d006d92320 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:04 ams-n28-06-12 scylla[4036370]:  [shard  6:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d00a88dd20 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard  7:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=7, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d006d92320 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard  5:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=5, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d006d92320 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard  6:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=6, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d00a88dd20 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard  3:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d00a88f220 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard  3:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=3, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d00a88f220 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard 13:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d00dede620 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard 13:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=13, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d00dede620 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard  0:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d00dede620 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard  0:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=0, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d00dede620 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard 10:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d00a89bb20 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard 10:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=10, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d00a89bb20 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard 40:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d00a88d420 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:05 ams-n28-06-12 scylla[4036370]:  [shard 40:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=40, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d00a88d420 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard  2:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d0077c2620 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard  2:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=2, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d0077c2620 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard 47:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d00a89bb20 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard 47:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=47, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d00a89bb20 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard 12:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d003317520 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard  4:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d00320ce20 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard  4:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=4, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d00320ce20 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard 12:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=12, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d003317520 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard  1:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d00320e920 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard 44:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d00a89a620 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard  1:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: shard=1, keyspace=accounts, cf=refresh_token, range=(-8978785552260999457,-8976503553854493619], got error in row level repair: seastar::rpc::remote_verb_error ([validator 0x61d00320e920 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
cze 12 05:53:06 ams-n28-06-12 scylla[4036370]:  [shard 45:stre] repair - repair[da893cb2-5ede-4eff-8292-af2429840d9e]: get_sync_boundary: got error from node=10.210.20.83, keyspace=accounts, table=refresh_token, range=(-8978785552260999457,-8976503553854493619], error=seastar::rpc::remote_verb_error ([validator 0x61d003315120 for mutation_compactor for compaction (accounts.refresh_token f9bd3b10-2512-11ec-8ff4-000000000018)] invalid partition-end, partition  has an active range tombstone {tombstone: timestamp=1716773420495561, deletion_time=1716773420})
```

Repair can't be done because of that. Please advise what to do.

**Comment by michoecho** [comment]
There were similar reports in the past and AFAIK it was judged that these errors are false negatives in the validator, which was never supposed to be enabled for this code path. And because of that, it was disabled in version 5.4.7 (by https://github.com/scylladb/scylladb/pull/18724).

So IIUC the recommendation here would be to upgrade from 5.4.6 to 5.4.7.

@denesb Please validate the above.

**Comment by denesb** [comment]
> There were similar reports in the past and AFAIK it was judged that these errors are false negatives in the validator, which was never supposed to be enabled for this code path. And because of that, it was disabled in version 5.4.7 (by #18724).
> 
> So IIUC the recommendation here would be to upgrade from 5.4.6 to 5.4.7.
> 
> @denesb Please validate the above.

Correct, this issue should be fixed in 5.4.6. @r4fek please upgrade and let us know if it helped.

**Comment by r4fek** [comment]
Upgrading to 5.4.7 fixed this problem. Thank you!
